plugins/modules/systemd_creds_encrypt.py

```diff
@@ -115,27 +138,59 @@ def main():
     secret = module.params["secret"]
     timestamp = module.params["timestamp"]
     user = module.params["user"]
+    dest = module.params["dest"]
+
+    if dest and os.path.isfile(dest):
+        decrypt_cmd = [cmd, "decrypt"]
+        if name:
+            decrypt_cmd.append("--name=" + name)
+
+        decrypt_cmd.append(dest)
+        decrypt_cmd.append("-")
+
+        rc, stdout, stderr = module.run_command(decrypt_cmd)
+        if rc != 0:
+            return module.fail_json(
+                "The credential file already exists; Couldn't decrypt to verify if it contains the same secret",
+                changed=False,
+                value=None,
+                rc=rc,
+                stderr=stderr,
+            )
+        elif stdout == secret:
+            return module.exit_json(
+                changed=False,
+                value=None,
+                rc=rc,
+                stderr=stderr,
+            )
 
     encrypt_cmd = [cmd, "encrypt"]
-    if name:
+    if name is not None:
         encrypt_cmd.append("--name=" + name)
-    else:
+    elif not dest:
         encrypt_cmd.append("--name=")
     if not_after:
         encrypt_cmd.append("--not-after=" + not_after)
-    if pretty:
+    if pretty and not dest:
```

Comment:
since those options were marked mutually exclusive, if `pretty is True`, you are guaranteed not to have `dest` set, therefore this check is redundant.